### PR TITLE
openh264: refactor + add 2.1.1 release

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **openh264/2.1.1**

Hinted by @qchateau 's conan-center-bot because 1.7.0 was almost 3 years old.
https://qchateau.github.io/conan-center-bot/#/updatable

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by prince-chrismc — DISMISSED** [review]

**Review by theirix — APPROVED** [review]

File: recipes/openh264/all/conanfile.py [hunk 1]
```diff
@@ -6,120 +6,163 @@
...
+            tools.replace_in_file(os.path.join(self._source_subfolder, "build", "platform-msvc.mk"),
+                                "CFLAGS_DEBUG += -MTd -Gm",
+                                "CFLAGS_DEBUG += -%s -Gm" % str(self.settings.compiler.runtime))
+        if self.settings.os == "Android":
+                tools.replace_in_file(os.path.join(self._source_subfolder, "codec", "build", "android", "dec", "jni", "Application.mk"),
+                    "APP_STL := stlport_shared",
+                    "APP_STL := %s" % str(self.settings.compiler.libcxx))
+                tools.replace_in_file(os.path.join(self._source_subfolder, "codec", "build", "android", "dec", "jni", "Application.mk"),
+                    "APP_PLATFORM := android-12",
+                    "APP_PLATFORM := %s" % self._android_target)
```
Comment (uilianries) [inline recipes/openh264/all/conanfile.py:63]:
```suggestion
        if self.settings.os == "Android":
            tools.replace_in_file(os.path.join(self._source_subfolder, "codec", "build", "android", "dec", "jni", "Application.mk"),
                                  "APP_STL := stlport_shared",
                                  "APP_STL := %s" % str(self.settings.compiler.libcxx))
            tools.replace_in_file(os.path.join(self._source_subfolder, "codec", "build", "android", "dec", "jni", "Application.mk"),
                                  "APP_PLATFORM := android-12",
                                  "APP_PLATFORM := %s" % self._android_target)
```
Bad indentation

Comment (madebr) [reply recipes/openh264/all/conanfile.py:63]: image

**Review by prince-chrismc — APPROVED** [review]

**Review by theirix — APPROVED** [review]

**Comment by madebr** [comment]
@uilianries 
This pr is missing your stamp of approval

**Review by SSE4 — APPROVED** [review]

File: recipes/openh264/all/conanfile.py [hunk 2]
```diff
@@ -6,120 +6,163 @@
...
         if self.settings.os == "Android":
             self.cpp_info.system_libs.append("m")
-        libcxx = self.settings.get_safe("compiler.libcxx")
-        if libcxx in ["libstdc++", "libstdc++11"]:
-            self.cpp_info.system_libs.append("stdc++")
-        elif libcxx == "libc++":
-            self.cpp_info.system_libs.append("c++")
-        elif libcxx in ["c++_static", "c++_shared"]:
-            self.cpp_info.system_libs.extend([libcxx, "c++abi"])
+        self.cpp_info.names["pkg_config"] = "libopenh264"
```
Comment (ericLemanissier) [inline recipes/openh264/all/conanfile.py:165]: @madebr Why this change ? it does not seem conform to upstream https://github.com/cisco/openh264/blob/master/openh264.pc.in

**Review by uilianries — COMMENTED** [review]
LGTM

**Comment by prince-chrismc** [comment]
Approving a PR that was merged! 🤣 

Thank god 2020 is over!!!!!!

**Comment by uilianries** [comment]
I was late for the party, but not for my vows of faith.
